Basic-Video-Capturer-Camera-2-Java/app/src/main/java/com/tokbox/sample/basicvideocapturercamera2/MirrorVideoCapturer.java

```diff
@@ -369,93 +381,121 @@ private int startCameraCapture() {
      */
     @Override
     public synchronized int startCapture() {
-        Log.d(TAG,"startCapture enter (cameraState: "+ cameraState +")");
+        Log.d(TAG,"startCapture() enter (cameraState: " + cameraState + ")");
+        Runnable resume = () -> {
+            initCamera();
+            scheduleStartCapture();
+        };
+        if (cameraState == CameraState.CLOSING) {
+            executeAfterClosed = resume;
+        } else if (cameraState == CameraState.CLOSED) {
+            resume.run();
+        } else {
+            scheduleStartCapture();
+        }
+        Log.d(TAG,"startCapture() exit");
+        return 0;
+    }
 
+    /**
+     * Starts capturing video.
+     */
+    public synchronized int scheduleStartCapture() {
+        Log.d(TAG,"scheduleStartCapture() enter (cameraState: " + cameraState + ")");
         if (null != camera && CameraState.OPEN == cameraState) {
-            return startCameraCapture();
+            return doStartCapture();
         } else if (CameraState.SETUP == cameraState) {
-            Log.d(TAG,"camera not yet ready, queuing the start until camera is opened");
-            executeAfterCameraOpened = () -> startCameraCapture();
-        } else {
-            throw new Camera2Exception("Start Capture called before init successfully completed");
+            Log.d(TAG,"camera not yet ready, queuing the start until camera is opened.");
+            executeAfterCameraOpened = this::doStartCapture;
+        } else if (CameraState.CREATESESSION == cameraState) {
+            Log.d(TAG,"Camera session creation already requested");
         }
-
-        Log.d(TAG,"startCapture exit");
-
+        else {
+            Log.d(TAG,"Start Capture called before init successfully completed.");
+        }
+        Log.d(TAG,"scheduleStartCapture() exit");
         return 0;
     }
 
-    /*
+    /**
      * Stops capturing video.
      */
     @Override
     public synchronized int stopCapture() {
         Log.d(TAG,"stopCapture enter");
-
         if (null != camera && null != captureSession && CameraState.CLOSED != cameraState) {
             cameraState = CameraState.CLOSING;
             try {
                 captureSession.stopRepeating();
             } catch (CameraAccessException e) {
                 e.printStackTrace();
             }
-
             captureSession.close();
-            cameraFrame.close();
-            characteristics = null;
-        }
+            cameraInfoCache = null;
+        } else if (null != camera && CameraState.OPEN == cameraState) {
+            cameraState = CameraState.CLOSING;
+            camera.close();
+        } else if (CameraState.SETUP == cameraState) {
+            /*
+            executeAfterCameraOpened = () -> {
+                cameraState = CameraState.CLOSING;
+                if (camera != null) {
+                    camera.close();
+                }
+            };
 
+             */
+            executeAfterCameraOpened = null;
```

Comment:
If  I remember well, this code should not be commented out, and the null assignment removed.

Reply:
I think the same. Not sure why this got commented out. May be we can check the history when this change was made.